File: app/src/main/java/pjurado/com/ej0102/MainActivity.java
package pjurado.com.ej0102;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private Button btn1;
    private Button btn2;
    private Button btn3;
    private Button btn4;
    private TextView tv1;
    private TextView tv2;
    private TextView tv3;
    private TextView tv4;


    private final static int UNO = 1;
    private final static int DOS = 2;
    private final static int TRES = 3;
    private final static int CUATRO = 4;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btn1 = findViewById(R.id.button1);
        btn2 = findViewById(R.id.button2);
        btn3 = findViewById(R.id.button3);
        btn4 = findViewById(R.id.button4);

        tv1 = findViewById(R.id.textView1);
        tv2 = findViewById(R.id.textView2);
        tv3 = findViewById(R.id.textView3);
        tv4 = findViewById(R.id.textView4);

        btn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, ActivitySecundario.class);
                i.putExtra("Valor", Integer.parseInt(tv1.getText().toString()));
                startActivityForResult(i, UNO);
            }
        });

        btn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, ActivitySecundario.class);
                i.putExtra("Valor", Integer.parseInt(tv2.getText().toString()));
                startActivityForResult(i, DOS);
            }
        });

        btn3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, ActivitySecundario.class);
                i.putExtra("Valor", Integer.parseInt(tv3.getText().toString()));
                startActivityForResult(i, TRES);
            }
        });

        btn4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, ActivitySecundario.class);
                i.putExtra("Valor", Integer.parseInt(tv4.getText().toString()));
                startActivityForResult(i, CUATRO);
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        int contador;

        if (resultCode == RESULT_OK) {
            switch (requestCode) {
                case UNO:
                    contador = Integer.parseInt(tv1.getText().toString()) + 1;
                    tv1.setText(String.valueOf(contador));
                    break;
                case DOS:
                    contador = Integer.parseInt(tv2.getText().toString()) + 1;
                    tv2.setText(String.valueOf(contador));
                    break;
                case TRES:
                    contador = Integer.parseInt(tv3.getText().toString()) + 1;
                    tv3.setText(String.valueOf(contador));
                    break;
                case CUATRO:
                    contador = Integer.parseInt(tv4.getText().toString()) + 1;
                    tv4.setText(String.valueOf(contador));
                    break;
            }

        }
    }
}

File: app/src/main/java/pjurado/com/ej0102/MainActivity.kt
package pjurado.com.ej0102

import androidx.appcompat.app.AppCompatActivity
import androidx.activity.result.contract.ActivityResultContracts
import android.widget.TextView
import android.os.Bundle
import pjurado.com.ej0102.R
import android.content.Intent
import pjurado.com.ej0102.ActivitySecundario
import pjurado.com.ej0102.MainActivity
import android.app.Activity
import android.view.View
import android.widget.Button
import com.google.android.material.snackbar.Snackbar
import pjurado.com.ej0102.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    companion object {
        private const val UNO = 1
        private const val DOS = 2
        private const val TRES = 3
        private const val CUATRO = 4
    }

    private lateinit var binding: ActivityMainBinding
    private val responseLauncher = registerForActivityResult(
        ActivityResultContracts.StartActivityForResult()){
        val requestCode = it.data?.extras?.getInt("requestCode")
        when (it.resultCode) {
            RESULT_OK -> {
                when (requestCode) {
                    UNO -> {
                        val contador = binding.textView1.text.toString().toInt() + 1
                        binding.textView1.text = contador.toString()
                    }
                    DOS -> {
                        val contador = binding.textView2.text.toString().toInt() + 1
                        binding.textView2.text = contador.toString()
                    }
                    TRES -> {
                        val contador = binding.textView3.text.toString().toInt() + 1
                        binding.textView3.text = contador.toString()
                    }
                    CUATRO -> {
                        val contador = binding.textView4.text.toString().toInt() + 1
                        binding.textView4.text = contador.toString()
                    }
                    RESULT_CANCELED -> Snackbar.make(
                        binding.root,
                        "Ha cancelado",
                        Snackbar.LENGTH_SHORT
                    )
                        .setBackgroundTint(resources.getColor(R.color.design_default_color_secondary_variant))
                        .show()
                }
            }
        }
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.button1.setOnClickListener(View.OnClickListener {
            val intent = Intent(this, ActivitySecundario::class.java)
            intent.putExtra("requestCode", UNO)
            intent.putExtra("Valor", binding.textView1.getText().toString().toInt())
            responseLauncher.launch(intent)
        })

        binding.button2.setOnClickListener(View.OnClickListener {
            val intent = Intent(this, ActivitySecundario::class.java)
            intent.putExtra("requestCode", DOS)
            intent.putExtra("Valor", binding.textView2.getText().toString().toInt())
            responseLauncher.launch(intent)
        })

        binding.button3.setOnClickListener(View.OnClickListener {
            val intent = Intent(this, ActivitySecundario::class.java)
            intent.putExtra("requestCode", TRES)
            intent.putExtra("Valor", binding.textView3.getText().toString().toInt())
            responseLauncher.launch(intent)
        })

        binding.button4.setOnClickListener(View.OnClickListener {
            val intent = Intent(this, ActivitySecundario::class.java)
            intent.putExtra("requestCode", CUATRO)
            intent.putExtra("Valor", binding.textView4.getText().toString().toInt())
            responseLauncher.launch(intent)
        })
    }

}